packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,295 @@
+          } = task;
+
+          return {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+            isComplete: task.lastCompleted != null,
+          };
+        },
+      );
+      return {
+        tasks: updatedTasksDto,
+      };
+    }
+  }
+
+  @Delete('/:id')
+  @HttpCode(204)
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id for the task to delete',
+  })
+  @ApiOperation({ summary: 'delete a task from a house.' })
+  @ApiNoContentResponse({
+    description: 'task successfully deleted.',
+  })
+  @ApiForbiddenResponse({
+    description: 'user is not the owner of the house',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async deleteTaskFromHouse(@Param('id') id, @User() user: DecodedIdToken) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const task = await this.taskStoreService.findOne(id);
+    if (!task) {
+      throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+    }
+
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException(
+        'user is not in the house',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (!house.owner.equals(userDoc._id)) {
+      throw new HttpException(
+        'user is not the owner of the house',
+        HttpStatus.FORBIDDEN,
+      );
+    }
+
+    this.taskStoreService.delete(task._id);
+  }
+
+  @Put('/:id/completed')
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id for the task to mark as complete or not',
+  })
+  @ApiOperation({
+    summary: 'mark user as having completed/not completed task.',
+  })
+  @ApiResponse({
+    description: 'tasks successfully marked as complete/not complete.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not assigned to the task',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async markTaskAsComplete(
+    @Body() completeTaskDto: CompleteTaskDto,
+    @Param('id') id,
+    @User() user: DecodedIdToken,
+  ) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const task = await this.taskStoreService.findOne(id);
+    if (!task) {
+      throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+    }
+
+    if (task.assigned !== user.uid) {
+      throw new HttpException(
+        'user is not assigned to task',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    if (completeTaskDto.isComplete) {
+      this.taskStoreService.update(task._id, { lastCompleted: new Date() });
+    } else {
+      this.taskStoreService.update(task._id, { lastCompleted: null });
+    }
+  }
```

Comment:
Personally I would make the update object the subject of the conditional here so that it doesn't look like we are conditionally updating, instead it looks like we are definitely updating, but using the conditional to change the update value.
`lastComplete = completeTaskDto.isComplete ? new Date() : null`

Reply:
minimising branching logic is good in my book but others might feel differently